lib/text_diff.ml

```diff
@@ -542,9 +541,11 @@ and process_arrow_type_diff ~context arrow_diff =
         return_type_ihunks;
       ]
   in
-  if context = `Tuple || context = `Larrow then
-    (Icommon "(" :: arrow_hunks) @ [ Icommon ")" ]
-  else arrow_hunks
+  match context with
+  | `Tuple | `Larrow -> parenthesize arrow_hunks
+  | _ -> arrow_hunks
```

Comment:
This case should be explicit, if we use a wildcard, we won't get warned by the compiler if we pass a new context down the chain and might forget to handle it correctly here! Same for the one above!